### ISSUE TITLE
[HOLD for payment 2022-04-13] [$250] When you click on a message and then press SHIFT a blue border appears - reported by @mdneyazahmad

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#7853]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to any chat 
2. click on any message and press SHIFT 

## Expected Result:
Border should not appear. 

**Remove the behavior where shift + click makes the message appear to be tabbed in focus.**

## Actual Result:
Blue border appears 

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- Desktop App

**Version Number:** 1.1.39-1
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

https://user-images.githubusercontent.com/43995119/155061030-62e2df79-6616-44e6-981b-8febe898e739.mov

**Expensify/Expensify Issue URL:**
**Issue reported by:** @mdneyazahmad
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1644402201151769

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

posted [here](https://www.upwork.com/jobs/~01022524387d167a0e)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @johncschuster (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @pecanoro (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @michelle-thompson (`Design`), see [these Stack Overflow questions](https://stackoverflow.com/c/expensify/search?q=%5Bchore%5D%5Bteam-design%5D+is%3Aquestion&searchOn=1) for more details.

**Comment by pecanoro** [comment]
@michelle-thompson @shawnborton  tagging you both before I assign external to verify what's the desired behaviour here, we don't want that blue border at all, right?

**Comment by parasharrajat** [comment]
That blue border indicates the focused item via Tab navigation.

**Comment by pecanoro** [comment]
@parasharrajat Ohh gotcha, but the behavior is a little buggy, isn't it? Like when you hover another element it does disappear only partially.

**Comment by parasharrajat** [comment]
I agree that is buggy indeed. But I would say that should be fixed for all elements in the app that have this issue. Also, this should be  task under accessibility.


But IMO this issue is irrelevant to that bug.

**Comment by shawnborton** [comment]
Yeah I think from an accessibility perspective, we want the blue outline when a message is "selected" to show a visual cue that the message item is in focus - but that being said, I would think the only way to select a message is to tab through the items. So I think maybe we should just get rid of this select + click behavior. Thoughts?

**Comment by michelle-thompson** [comment]
Yup, agree with Shawn here!

**Comment by MelvinBot** [comment]
@michelle-thompson Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by pecanoro** [comment]
I am a bit confused so do we want to remove the tab functionality completely?

**Comment by shawnborton** [comment]
I don't think we want to remove the tab functionality. I'm just suggesting we remove that weird behavior where shift + click makes the message appear to be tabbed in focus.

**Comment by pecanoro** [comment]
@shawnborton Awesome, thanks! I updated the description to add that so when we export the issue, it's clear what we want to accomplish.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jboniface (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by jboniface** [comment]
posted [here](https://www.upwork.com/jobs/~01022524387d167a0e)

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`Exported`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @tgolen (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by phivh** [comment]
**Proposal:**
Remove the box-shadow in 
https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/web/index.html#L35-L37

```
:focus-visible {
   outline: 0;
 }
```
Demo: 

https://user-images.githubusercontent.com/5449119/156870304-b111787b-8e78-4837-a27a-2ab9c4517ac3.mp4

**Comment by mollfpr** [comment]
## Proposal

Adding event listener to type`keydown` and call `blur()` to unfocus the component. In this way, the keyboard accessibility still working. We can add a prop to determine should this function is activated or not.

```
this.pressableRef.addEventListener('keydown', () => {
    this.pressableRef.blur();
});
```

https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/src/components/PressableWithSecondaryInteraction/index.js#L18-L23

## Result

https://user-images.githubusercontent.com/25520267/156880097-475e2450-8346-4cce-9c2a-197d7f74ad0c.mov

**Comment by sokolenkov** [comment]
I can easilly fix this issue without adding any JS code into the project. Contact me, if this issue is still appears

**Comment by rushatgabhane** [comment]
@sokolenkov how do you plan to fix this issue?
I'd suggest that you read https://github.com/Expensify/App/blob/d4ed740be06862eb4ea31fa831de9416f7ed0a55/CONTRIBUTING.md before posting a proposal, thanks!

**Comment by sokolenkov** [comment]
@rushatgabhane I need access to the main styles file, where I need to make small change. Hasn't this issue been fixed by previous contributors yet?

**Comment by rushatgabhane** [comment]
🎀 👀 🎀 C+ reviewed
@mollfpr's proposal looks good.

cc: @tgolen

**Comment by rushatgabhane** [comment]
> I need access to the main styles file, where I need to make small change. Hasn't this issue been fixed by previous contributors yet?

Yeah sorry about that, it's fixed by @mollfpr

**Comment by rushatgabhane** [comment]
> the behavior in this issue not only appear to click + shift but also to click + another key event.


@mollfpr your proposal should fix the issue for all key events and `Tab` would work as expected.
But it seems like a hack to me

**Comment by rushatgabhane** [comment]
@phivh thanks, but your proposal doesn't work for other keys like any alphabet

**Comment by mollfpr** [comment]
@rushatgabhane I have another proposal

Adding this code will unfocs the the `ReportActionItem` when selected and pressing the keydown. The Tab accessbility still working. This approach more cleaner than my proposal before.
```js
onKeyDown={(e) => {
    e.target.blur();
}}
```
https://github.com/Expensify/App/blob/afaccd9af40cfc2df1a6b12abd4234183975e476/src/pages/home/report/ReportActionItem.js#L151-L157

https://user-images.githubusercontent.com/25520267/157640253-1c8f2dfe-de06-4e22-9582-5c5ec70654f2.mov

**Comment by rushatgabhane** [comment]
@mollfpr can you think of a solution where the blue border doesn't appear in the first place?
Letting it focus, and blurring immediately seems like a hack to me.

**Comment by mollfpr** [comment]
@rushatgabhane The blue border appear from attribute `data-focusvisible-polyfill` and this set from keydown event. As you can see from the screenshot, there's a file called `modality.js` from the `react-native-web` package.

<img width="638" alt="Screen Shot 2022-03-11 at 21 11 45" src="https://user-images.githubusercontent.com/25520267/157885313-438823ed-1d81-495c-b863-6c87fca96c16.png">
<img width="556" alt="Screen Shot 2022-03-11 at 21 21 42" src="https://user-images.githubusercontent.com/25520267/157885612-774c178d-7bf0-4cd3-974c-2ebdcd0b1b06.png">

**Comment by rushatgabhane** [comment]
@mollfpr nice! I see that it's returning early if `Tab` isn't pressed. So why are we still getting the blue border after pressing any letter?

**Comment by mollfpr** [comment]
@rushatgabhane I think because the second condition `(e.metaKey || e.altKey || e.ctrlKey || e.shiftKey)` returning `false`.

I think there's no other way except a hacky way to solve this issue. Everything is done by `react-native-web`.

**Comment by mollfpr** [comment]
@rushatgabhane Nope, I checked the code modality.js from react-native-web and try to change the code. Also the condition are *if not Tab and is metakey or shiftkey or ctrkey or altkey*.

**Comment by rushatgabhane** [comment]
You're right, thanks for correcting me!

**Comment by rushatgabhane** [comment]
@mollfpr we also need to make sure that `Shift + Tab` works too. Can you please confirm that works too?

**Comment by mollfpr** [comment]
@rushatgabhane yup still working.

https://user-images.githubusercontent.com/25520267/159423095-38ae9c6b-9dec-4f09-9ab4-0badb7403a79.mov

**Comment by rushatgabhane** [comment]
🎀 👀 🎀 C +reviewed
@tgolen I recommend @mollfpr's [proposal ](https://github.com/Expensify/App/issues/7853#issuecomment-1063887115) . It's a workaround but I doubt that we can get it fixed at the root level https://github.com/Expensify/App/issues/7853#issuecomment-1065209149

**Comment by mollfpr** [comment]
@tgolen Sorry not to make it clear. The `onKeyDown` will be pass to `PressableWithSecondaryInteraction`. Basically, the focus blue border is set from `react-native-web` by the keyboard event listener, it's added `data-focusvisible-polyfill="true"` and in our css will be set blue outline around the element.

So the solution I propose is we blur immediately when the key is pressed, in this way we still manage to keep the border from accessibility `Tab` or `Shift + Tab`.

https://github.com/Expensify/App/blob/afaccd9af40cfc2df1a6b12abd4234183975e476/src/pages/home/report/ReportActionItem.js#L151-L157
```diff
 <PressableWithSecondaryInteraction 
     ref={el => this.popoverAnchor = el} 
     onPressIn={() => this.props.isSmallScreenWidth && canUseTouchScreen() && ControlSelection.block()} 
     onPressOut={() => ControlSelection.unblock()} 
     onSecondaryInteraction={this.showPopover} 
     preventDefaultContentMenu={!this.props.draftMessage} 
+   onKeyDown={(e) => {
+        e.target.blur();
+    }}
```

**Comment by jboniface** [comment]
@tgolen bump for addressing proposal

**Comment by tgolen** [comment]
Thanks for the bump! I missed this when I was OOO.

Thanks @mollfpr for clarifying that! I think it's an OK solution, but it looks like `PressableWithSecondaryInteraction` is used in 4 other places in the code. Should the solution then be to add `onKeyDown` to all 5 places, only in this 1 place, or to do it instead inside the component (so it's applied to all 5 places automatically)?

**Comment by mollfpr** [comment]
I think is better to pass `onKeyDown` only in this 1 place.

Currently `PressableWithSecondaryInteraction` is only use by `ReportActionItem` and `AnchorForCommentsOnly`. For `AnchorForCommentsOnly` is also rendered inside `ReportActionItem` so it's already covered.

But I'm up for others' input.

cc @tgolen

**Comment by tgolen** [comment]
Ah, OK. Makes sense, thank you! 🟢  to hire @mollfpr for the proposal @jboniface

**Comment by MelvinBot** [comment]
📣 @mollfpr You have been assigned to this job by @jboniface!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md) 📖

**Comment by jboniface** [comment]
@mollfpr have you applied to the Upwork job?

**Comment by mollfpr** [comment]
@jboniface not yet, I couldn't find the link.

**Comment by jboniface** [comment]
@mdneyazahmad hey, can you apply [here](https://www.upwork.com/jobs/~01022524387d167a0e)? you're supposed to get a reporting bonus for this

**Comment by mdneyazahmad** [comment]
@jboniface I already applied. Thank you

**Comment by jboniface** [comment]
paid everyone

**Comment by parasharrajat** [comment]
Because of this issue's PR, we are facing a new issue https://github.com/Expensify/App/issues/12554. 

Is it really important to fix this blue border behavior? I checked with chrome and Chrome by default does that. There are complex solutions to fix the new bug but I still think the solution in [PR](https://github.com/Expensify/App/pull/8443) is very unstable. It can have unknown side-effects. 

My suggestion is to revert this change and keep this blue border behavior. 

@shawnborton https://github.com/Expensify/App/issues/7853#issuecomment-1058632633 Do you feel that we need to fix this issue? 

cc: @rushatgabhane

**Comment by shawnborton** [comment]
I would be fine with your suggestion @parasharrajat

**Comment by s77rt** [comment]
+1 for the revert

**Comment by s77rt** [comment]
I think my proposal here https://github.com/Expensify/App/issues/12010#issuecomment-1312059021 would be better than a revert. It fixes the original issue and does not cause weird behaviour.

**Comment by parasharrajat** [comment]
This will be reverted as part of https://github.com/Expensify/App/issues/12554. We have found a better solution.
